lms/djangoapps/course_api/blocks/views.py

```diff
@@ -0,0 +1,263 @@
+            Returned only if "format" is included in the "requested_fields" parameter.
+
+          * student_view_data: (dict) The JSON data for this block.
+            Returned only if the "student_view_data" input parameter contains this block's type.
+
+          * student_view_url: (string) The URL to retrieve the HTML rendering of this block's student view.
+            The HTML could include CSS and Javascript code. This field can be used in combination with the
+            student_view_multi_device field to decide whether to display this content to the user.
+
+            This URL can be used as a fallback if the student_view_data for this block type is not supported by
+            the client or the block.
+
+          * student_view_multi_device: (boolean) Whether or not the block's rendering obtained via block_url has support
+            for multiple devices.
+            Returned only if "student_view_multi_device" is included in the "requested_fields" parameter.
+
+          * lms_web_url: (string) The URL to the navigational container of the xBlock on the web LMS.
+            This URL can be used as a further fallback if the student_view_url and the student_view_data fields
+            are not supported.
+
+    """
+    def list(self, request, usage_key_string):
+        """
+        REST API endpoint for listing all the blocks and/or navigation information in the course,
+        while regarding user access and roles.
+
+        Arguments:
+            request - Django request object
+            course - course module object
+        """
+
+        # validate request parameters
+        requested_params = request.QUERY_PARAMS.copy()
+        requested_params.update({'usage_key': usage_key_string})
+        params = BlockListGetForm(requested_params, initial={'requesting_user': request.user})
+        if not params.is_valid():
+            raise ValidationError(params.errors)
+
+        try:
+            return Response(
+                get_blocks(
+                    request,
+                    params.cleaned_data['usage_key'],
+                    params.cleaned_data['user'],
+                    params.cleaned_data.get('depth', None),
+                    params.cleaned_data.get('nav_depth', None),
+                    params.cleaned_data['requested_fields'],
+                    params.cleaned_data.get('block_counts', []),
+                    params.cleaned_data.get('student_view_data', []),
+                    params.cleaned_data['return_type']
+                )
+            )
+        except ItemNotFoundError as exception:
+            raise Http404("Block not found: {}".format(exception.message))
+
+
+@view_auth_classes()
+class BlocksInCourseView(BlocksView):
+    """
+    **Use Case**
+
+        Returns the blocks in the course according to the requesting user's access level.
+
+    **Example requests**:
+
+        GET /api/courses/v1/blocks/?course_id=<course_id>
```

Comment:
The endpoint below includes all the indented lines prefixed with `&`, so they're not the same, unless I'm missing what you mean by "below."